actions/signed-commits/src/git/github-git/index.ts

```diff
@@ -21,30 +23,56 @@ export async function commitAll(
   message: string,
   cwd?: string,
 ) {
-  const input: CreateCommitOnBranchInput = {
-    branch: {
-      branchName: branch,
-      repositoryNameWithOwner: `${owner}/${repo}`,
-    },
-    message: {
-      headline: message,
-      body: "",
-    },
-    expectedHeadOid: await getRemoteHeadOid(client, {
-      branch,
-      owner,
-      repo,
-    }),
-    fileChanges: await repoStatus.getFileChanges(cwd),
-  };
-
-  await createCommitOnBranch(client, input);
+  core.debug(`Committing all changes in ${cwd} to ${owner}/${repo}:${branch}`);
+
+  const fileChanges = await repoStatus.getFileChanges(cwd);
+  const commits = compileCommits(fileChanges, message);
+
+  if (commits.length === 0) {
+    core.info("No changes to commit. Skipping.");
+  }
+
+  let expectedHeadOid = await getRemoteHeadOid(client, {
+    branch,
+    owner,
+    repo,
+  });
+
+  let commitNumber = 1;
+  core.info(`Creating ${commits.length} commits.`);
+  for (const commit of commits) {
+    core.debug(
+      `Commit ${commitNumber++}/${commits.length} - ${commit.message}`,
+    );
+    const input = {
+      branch: {
+        branchName: branch,
+        repositoryNameWithOwner: `${owner}/${repo}`,
+      },
+      message: {
+        headline: commit.message,
+        body: "",
+      },
+      expectedHeadOid,
+      fileChanges: commit.fileChanges,
+    };
+    const response = await createCommitOnBranch(client, input);
+    expectedHeadOid = getRemoteHeadOidFromResponseOrThrow(response);
```

Comment:
If this errors mid-way, is it possible that we get stuck in a bad state that would require manual intervention vs being able to re-run the action?

The current edge case I can think of is:

- A transient network error occurs between 2 commits, one deletion commit followed by one addition commit 
- If the action is re-run, we may miss updating changelogs / package.json's since the changesets no longer exist

Reply:
This commit applies to this entire loop now that I think of it, we need to make sure all changes are applied in serial order and can be recovered from in case of error

Reply:
I believe rerunning the workflow would solve the problem wouldn’t it?

Reply:
If the workflow is re-ran after the addition commit was already pushed but the deletion commit failed, it sounds like it'll create a commit conflict or miss updating changelogs / package.jsons

Reply:
https://github.com/smartcontractkit/.github/blob/da1c0099f3bab236660ccc10328a29e06950c69f/actions/signed-commits/src/run.ts#L341-L360